circle.yml

```diff
@@ -7,8 +7,24 @@ general:
 
 checkout:
   post:
-    # Update the `master` branch. Otherwise CircleCI will give us a cached one.
-    - git fetch -u origin +master:master
+    - |
+        # Add `master` branch to the update list.
+        # Otherwise CircleCI will give us a cached one.
+        FETCH_REFS="+master:master"
+        # Update PR refs for testing.
+        if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+        then
+            FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/head:pr/${CIRCLE_PR_NUMBER}/head"
+            FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/merge:pr/${CIRCLE_PR_NUMBER}/merge"
+        fi
+        # Retrieve the refs
+        eval "git fetch -u origin ${FETCH_REFS}"
```

Comment:
I think so (added this last night), but let me try removing it and see if it doesn't just work.